src/logics/ArithLogic.h

```diff
@@ -95,6 +95,8 @@ class ArithLogic : public Logic {
 
     SymRef get_sym_Int_TIMES() const { return sym_Int_TIMES; }
     SymRef get_sym_Real_TIMES() const { return sym_Real_TIMES; }
+    SymRef get_sym_Int_TIMES_NONLIN() const { return sym_Int_TIMES_NONLIN; }
+    SymRef get_sym_Real_TIMES_NONLIN() const { return sym_Real_TIMES_NONLIN; }
```

Comment:
Personally I would use the terminology the other way around: 
```suggestion
    SymRef get_sym_Int_TIMES_LIN() const { return sym_Int_TIMES_LIN; }
    SymRef get_sym_Real_TIMES_LIN() const { return sym_Real_TIMES_LIN; }
```
because it the linear multiplication that is actually a special case, accepting only some arguments.

Reply:
It is a bit different

Both linear and nonlinear multiplication are special cases:

Nonlin is only of the view (* var_1, var_2, ..., var_n) - necessarily without any constants!

Lin is (* const var)

Reply:
What is e.g. `(* var const)` or `(* var const var)` then?